Add file cache headers feature flag and implement caching logic in fi…

txau opened the pull request:
…le routes

fixes #

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

Commit:
Add file cache headers feature flag and implement caching logic in file routes

Commit:
Update download route tests to use async/await for tenant changes and improve Last-Modified header assertions

Commit:
Refactor type definitions for consistency and clarity in commonTypes, connectionType, and settingsType files

konzz committed:
enabled feature by default for CI purposes

konzz committed:
updated entity e2e to dont wait for files

konzz committed:
public pages form fix

konzz committed:
Merge branch 'production' into cache-headers-files

konzz committed:
wrapped checkally to ensure axe is injected

konzz committed:
Merge branch 'cache-headers-files' of github.com:huridocs/uwazi into cache-headers-files

konzz committed:
fixed checkally command

konzz committed:
eslint fixes

konzz committed:
missing snapshots

konzz committed:
moved command to different file

konzz committed:
fixed eslint

konzz committed:
disabled flag by default

txau commented on app/api/files/routes.ts:
```diff
@@ -65,6 +67,54 @@ const filterByEntityPermissions = async (fileList: FileType[]): Promise<FileType
+      'published',
+      { withoutDocuments: true }
+    );
+
```
I am not sure about this try/catch block. Always weary of a catch block with silent behavior (although it returns false). What would be the logic and reason for not passing the try block?

daneryl replied:
We should not leave it like this, an error is an error, and transforming that error in a response like `false` is going to be a potential pain, let the error bubble up. 
In case there are specific known operational errors then conditionally catch them on the catch block.

txau commented on app/api/files/routes.ts:
```diff
@@ -221,15 +271,47 @@ export default (app: Application) => {
+      } catch (error) {
+        // If any cache header logic fails, silently skip it and continue serving file
+        // This ensures backwards compatibility and prevents blocking file access
+      }
```
Again a catch block failing silently

daneryl replied:
Yes, lets not leave it like this please

> // This ensures backwards compatibility and prevents blocking file access

What this makes sure of is to hide potential important errors, let the error bubble up, in this case it is probably enough to remove the try/catch block

Commit:
Merge remote-tracking branch 'origin/production' into cache-headers-files

Commit:
Refactor file upload handling and tenant model properties

- Updated file upload middleware to improve error handling and streamline the process.
- Removed deprecated tenant model properties and ensured consistency with the latest feature flags.

daneryl committed:
Isolate different file tests filePath

daneryl committed:
Merge branch 'production' into cache-headers-files

daneryl committed:
Add custom uploads testing path

daneryl committed:
Merge branch 'production' into cache-headers-files

daneryl committed:
Merge branch 'production' into cache-headers-files
